src/fsharp/FSComp.txt

```diff
@@ -13,6 +13,8 @@ undefinedNameRecordLabelOrNamespace,"The record label or namespace '%s' is not d
 undefinedNameRecordLabel,"The record label '%s' is not defined"
 undefinedNameTypeParameter,"The type parameter '%s' is not defined"
 undefinedNamePatternDiscriminator,"The pattern discriminator '%s' is not defined"
+missingElseBranch,"You have not supplied the \"else\" case for this expression. If / then is an expression and so it must always return a result of the same type in all cases. This expression was expected to have type '%s' but here has type '%s'."
```

Comment:
@dsyme I think this is slightly more concise and conveys the same meaning:
````
The 'if' expression is missing an 'else' branch, the 'then' branch has type '%s'. The 'else' branch must match this type, it may only be omitted when the 'then' branch has type 'unit'.
````
Failing test is 

[00:34:51] Libraries\Core\Unchecked (DefaultOf02)
[00:34:51]  -- passed
[00:34:51] Warnings (WrongNumericLiteral.fs)
[00:34:51]  -- passed
[00:34:51] Warnings (WarnIfMissingElseBranch.fs)
[00:34:51]  -- failed
[00:34:51] Warnings (ElseBranchHasWrongType.fs)
[00:34:51]  -- passed
[00:34:51] Libraries\Portable (consumer.fs (Desktop))